src/pybind/mgr/dashboard/frontend/src/app/shared/directives/form-input-disable.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { FormNameDirective } from './form-name.directive';
+
+@Directive({
+  selector:
+    'input:not([cdNoFormInputDisable]), select:not([cdNoFormInputDisable]), [cdFormInputDisable]'
+})
+export class FormInputDisableDirective implements AfterViewInit {
+  permissions: Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private service: FormNameDirective,
```

Comment:
This is not a service, we should use the directive name:
```suggestion
    @Optional() private formName: FormNameDirective,
```